smtp_by_api_yahoo.go

```diff
@@ -64,8 +68,23 @@ func (y yahoo) check(domain, username string) (*SMTP, error) {
 	}, nil
 }
 
+func getSessionIndex(resp *http.Response) (string, error) {
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	re := regexp.MustCompile(`value="([^"]+)" name="sessionIndex"`)
+
+	// 在响应体中查找匹配项
+	match := re.FindSubmatch(respBytes)
+	if len(match) > 1 {
+		return string(match[1]), nil
+	}
+	return "", errors.New("yahoo check by api, no sessionIndex")
+}
+
 func checkUsernameExists(resp *http.Response) (usernameExists bool, err error) {
-	respBytes, err := io.ReadAll(resp.Body)
+	respBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
`ioutil.ReadAll` is deprecated in favour of `io.ReadAll` (see [method description](https://pkg.go.dev/io/ioutil#ReadAll)).

Reply:
Ok, now I see that your commits were exploratory ones and "Files Changed" tab gives a very clear picture of changes.